core/Updater.php

```diff
@@ -226,7 +226,16 @@ public function getUpdateClassName($componentName, $fileVersion)
             return '\\Piwik\\Columns\\Updater';
         }
 
-        return '\\Piwik\\Plugins\\' . $componentName . '\\' . $className;
+        $backwardCompatibleClass = '\\Piwik\\Plugins\\' . $componentName . '\\' . $className;
+
+        // as of Piwik 3.0.0, we expect plugin updates within an Updates\ namespace
+        // but in Piwik 3.0.0 we stay backward compatible
+        // this could be @deprecated in Piwik 4.0.0
+        if(class_exists($backwardCompatibleClass, false)) {
```

Comment:
How does class_exists find the old class when autoload is disabled? Are they always first included in `loadComponentsWithUpdateFile()`? 

Can you add a test or something to remind us to remove this in Piwik 4.0 and to make sure it does not get removed earlier? Should be added to `DeprecatedMethodsTest` and can be likely eg done by moving this BC code into a separate method.